pkg/ui/src/views/statements/planView.tsx

```diff
@@ -16,115 +16,274 @@ import _ from "lodash";
+  const flattenedPlan = [ planNode(
+    treePlan.name,
+    treePlan.attrs,
+    []),
+  ];
+
+  if (treePlan.children.length === 0) {
+    return flattenedPlan;
+  }
+  const flattenedChildren =
+    treePlan.children
+      .map( (child) => flattenTree(child));
+  if (treePlan.children.length === 1) {
+    flattenedPlan.push(...flattenedChildren[0]);
+  } else {
+    flattenedPlan[0].children = flattenedChildren;
+  }
+  return flattenedPlan;
 }
 
-function PlanNode(props: PlanNodeProps) {
-  const node = props.node;
-  const collapsedAttrs = collapseRepeatedAttrs(node.attrs);
-  return (
-    <div>
-      <span className="plan-node__name">{node.name}</span>
-      <span className="plan-node__attrs">
-        {collapsedAttrs.map((attr) => (
-          <span className="plan-node__attr" key={attr.key}>
-            <span className="plan-node__attr-key">{attr.key}</span>
-            <span className="plan-node__attr-eq">=</span>
-            { renderAttrValueList(attr.value) }
-          </span>
-        ))}
-      </span>
-      { renderChildren(node.children) }
-    </div>
-  );
+interface IPlanNodeHeaderProps {
+  title: string;
+  subtitle: string;
+  warn: boolean;
 }
 
-function renderChildren(children: cockroach.sql.IExplainTreePlanNode[]|null): React.ReactElement<PlanNodeProps> {
-  if (!children) {
-    return null;
+// TODO(celia): explain.
+function planNodeHeaderProps(node: IPlanNode): IPlanNodeHeaderProps {
+  let title = _.capitalize(node.name);
+  let subtitle = null;
+  let warn = false;
+
+  switch (node.name) {
+    case "scan":
+      title = "Table scan";
+      if (_.find(node.attrs, {key: "spans", value: "ALL"})) {
+        warn = true;
+        const tableAttr = _.filter(node.attrs, (attr) => ( attr.key === "table"));
+        subtitle = tableAttr[0].value;
+        const ampersandPosition = subtitle.indexOf("@");
+        if (ampersandPosition > 0) {
+          subtitle = subtitle.substring(0, ampersandPosition - 1);
+        }
+      }
+      break;
+    case "join":
+      const typeAttr = _.filter(node.attrs, (attr) => ( attr.key === "type"));
+      title = _.capitalize(typeAttr[0].value + " " + node.name);
+      break;
+    default:
+      break;
   }
-  return (
-    <ul>
-      {children.map((child, idx) => (
-        <li>
-          <PlanNode key={idx} node={child} />
-        </li>
-      ))}
-    </ul>
-  );
+
+  return {
+    title, subtitle, warn,
+  };
 }
 
-function renderAttrValueList(values: string[]) {
-  return (
-    <span className="plan-node__attr-value-list">
-      [
-      {intersperse(
-        values.map((value, idx) => (
-          <span className="plan-node__attr-value" key={idx}>{value}</span>
-        )),
-        <span>, </span>,
-      )}
-      ]
-    </span>
-  );
+/* ************************* PLAN NODES ************************* */
+
+function PlanNodeDescription (props: {
+  header: IPlanNodeHeaderProps;
+}) {
+  return <span>
+    {props.header.warn &&
+    <span>&#x26A0; </span>
```

Comment:
What are these? Warning icons? Might want to add an explanatory comment, if you don't change to SVG.

Reply:
good call - this will eventually be replaced with SVG, but will add a comment for now